electrumsv/util/cache.py

```diff
@@ -101,15 +110,77 @@ def get(self, key: bytes) -> Optional[bytes]:
             self.misses += 1
         return None
 
-    def _resize(self) -> List[Tuple[bytes, bytes]]:
+    def _resize(self) -> List[Tuple[bytes, Transaction]]:
         removals = []
         while len(self._cache)-1 >= self._max_count or self.current_size > self._max_size:
             node = self._root.next
             previous_node, next_node, discard_key, discard_value = \
                 node.previous, node.next, node.key, node.value
             previous_node.next = next_node
             next_node.previous = previous_node
-            self.current_size -= len(discard_value)
+            self.current_size -= self.obj_size(discard_value)
             del self._cache[discard_key]
             removals.append((discard_key, discard_value))
         return removals
+
+    def obj_size(self, o):
```

Comment:
Can we make this a standalone functioon or does it need to be part of this class?

Reply:
Yeah - it should be stand-alone